plugins/inputs/logfile/logfile.go

```diff
@@ -107,8 +107,9 @@ func (t *LogFile) Start(acc telegraf.Accumulator) error {
 		return fmt.Errorf("failed to create state file directory %s: %v", t.FileStateFolder, err)
 	}
 
-	// Clean state file regularly
+	// Clean state file on init and regularly
 	go func() {
+		t.cleanupStateFolder()
```

Comment:
Can you add a unit test or integ test which creates some files monitored by the agent, write 1 MiB of logs to them, verify the state files point to the EOF, then kill the agent, then write more data, then start the agent back up and make sure they report logs events from the 1 MiB point onwards?

Maybe we already have a test like this, if so great.

My concern is that you are deleting the state files before the agent has a chance to start up and read them. (I think Zhihong has the same concern).